endpoints/openrtb2/auction.go

```diff
@@ -2352,6 +2357,29 @@ func setDoNotTrackImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrappe
 	}
 }
 
+func setCookieDeprecation(httpReq *http.Request, r *openrtb_ext.RequestWrapper, account *config.Account) {
+	if account == nil || !account.Auction.PrivacySandbox.CookieDeprecation.Enabled {
+		return
+	}
+
+	secCookieDeprecation := httpReq.Header.Get(secCookieDeprecation)
+	if secCookieDeprecation == "" {
+		return
+	}
+
+	if r.Device == nil {
+		r.Device = &openrtb2.Device{}
+	}
+
+	if deviceExt, err := r.GetDeviceExt(); err == nil {
+		ext := deviceExt.GetExt()
+		if ext["cdep"] == nil {
+			ext["cdep"] = json.RawMessage(fmt.Sprintf(`"%s"`, secCookieDeprecation))
+			deviceExt.SetExt(ext)
+		}
+	}
```

Comment:
This block of code should look something like this:
```
if deviceExt, err := r.GetDeviceExt(); err == nil {
    if cdep := deviceExt.GetCDep(); cdep == nil {
        deviceExt.SetCDep(cdep)
    }
}
```

You'll need to add cdep fields to `DeviceExt`, update the `marshal` and `unmarshal` methods and add the get and set function for cdep. See `RegExt.usprivacy` and `RegExt.gdpr` for reference.

All of the fields on `DeviceExt` are synced throughout the wrapper effectively updating the `DeviceExt.ext` and the wrapper bid request device ext when the wrapper `RebuildRequest` function is called in `HoldAuction`.

Reply:
done